### PR TITLE
A4A > Referrals: Initial implementation of automated referrals

### PR DESCRIPTION
**Pull request by yashwin**
Resolves https://github.com/Automattic/jetpack-genesis/issues/335
Resolves https://github.com/Automattic/jetpack-genesis/issues/340
Resolves https://github.com/Automattic/jetpack-genesis/issues/341

## Proposed Changes

This PR implements the initial changes required for the automated referrals. These changes are behind a feature flag.

NOTE: Please ignore the page background for now.

## Testing Instructions

1. Open the A4A live link.
2. Verify that you can see a chevron icon next to the Referrals

Screenshot 2024-05-21 at 7 22 14 PM

3. Click the Referrals > Verify that you can see sub-menu items as shown below. Also, verify the link works as expected.

Screenshot 2024-05-21 at 7 22 24 PM 

4. Verify that the feature flag is enabled only on Dev & Horizon
5. Append the /referrals URL as `/referrals?flags=-a4a-automated-referrals` & verify that the above changes are not reflected and the Referrals page matches with the production. 

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: config/a8c-for-agencies-development.json [hunk 1]
```diff
@@ -50,7 +51,8 @@
 		"a8c-for-agencies-signup": true,
 		"a8c-for-agencies-referrals": true,
 		"a8c-for-agencies-settings": true,
-		"a8c-for-agencies-migrations": true
+		"a8c-for-agencies-migrations": true,
+		"a8c-for-agencies-automated-referrals": true
```
Comment (jkguidaven) [inline config/a8c-for-agencies-development.json]: I don't see this PR introducing a new `a8c-for-agencies-automated-referrals` section. Is this correct?

**Review by jkguidaven — APPROVED** [review]
Works great. I found 1 line which I think was a mistake but is non-blocker.

Feel free to remove it before merging this.  👍

**Review by cleacos — APPROVED** [review]
LGTM! (see James' comment)

File: config/a8c-for-agencies-development.json (hunk 1 above)
Comment (cleacos) [reply config/a8c-for-agencies-development.json]: Good catch! It seems a mistake.

Comment (yashwin) [reply config/a8c-for-agencies-development.json]: Good catch! 

Yes, I missed removing this!

**Pull request merged**